samples/javascript_nodejs/52.logger-bot/bot.js

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { ActivityTypes } = require('botbuilder');
+
+// Turn counter property
+const TURN_COUNTER_PROPERTY = 'turnCounterProperty';
+
+class MyBot {
+  /**
+   *
+   * @param {ConversationState} conversation state object
+   */
+  constructor(conversationState) {
+    // Creates a new state accessor property.
+    // See https://aka.ms/about-bot-state-accessors to learn more about the bot state and state accessors.
+    this.countProperty = conversationState.createProperty(TURN_COUNTER_PROPERTY);
+    this.conversationState = conversationState;
+  }
+  /**
+   *
+   * @param {TurnContext} on turn context object.
+   */
+  async onTurn(turnContext) {
+    // See https://aka.ms/about-bot-activity-message to learn more about the message and other activity types.
+    if (turnContext.activity.type === ActivityTypes.Message) {
+      // read from state.
+      let count = await this.countProperty.get(turnContext);
+      count = count === undefined ? 1 : ++count;
+      await turnContext.sendActivity(`${count}: You said "${turnContext.activity.text}"`);
+      // increment and set turn counter.
+      await this.countProperty.set(turnContext, count);
+    } else {
+      await turnContext.sendActivity(`[${turnContext.activity.type} event detected]`);
+    }
+    // Save state changes
+    await this.conversationState.saveChanges(turnContext);
```

Comment:
Basic bot might be a better sample to base this off rather than the echo bot so we can actually show capturing transcripts for multi-turn conversations.

Reply:
For a similar bot to build off of, the [multi-turn-prompt](https://github.com/Microsoft/BotBuilder-Samples/tree/master/samples/javascript_nodejs/05.multi-turn-prompt) sample would probably also do nicely.

Reply:
Switched the bot over to the multi-turn-prompt bot for the moment since it doesn't require the services the basic bot requires.